### PR TITLE
Adds on_varedit proc

### PR DESCRIPTION
**Pull request by Jordie0608**
Currently not used but intended to improve the handling in #13153 for when/if that/this gets merged.

Mass modify is ugly yes, but that's how it works.

### CODE REVIEW

**Comment by YotaXP** [comment]
Preeeetty sure this should be declared on `/datum`.

**Comment by Jordie0608** [comment]
Yeah I just figured that myself and am changing it.

**Comment by YotaXP** [comment]
If I might propose an alternate signature, how about:
``` C
/datum/proc/onVarEdit(varName, newValue)
    vars[varName] = newValue
```
Then we can prevent sensitive values from changing, or handle the change ourselves on a case by case basis. (Removing snowflake such as that special case for "luminosity".)

Of course, then you'd have to change how it gets called.
``` C
A.onVarEdit(variable, new_value)
```
Then again, this has a problem with list variables...  meh, food for though.

**Comment by Aranclanos** [comment]
can you make a if(!O.on_varedit()) return
and make the parent return 1

File: code/modules/admin/verbs/massmodvar.dm [hunk 1]
```diff
@@ -453,6 +453,38 @@
...
+
+		else if(istype(O,/obj))
+			for(var/obj/A in world)
+				if(istype(A,O.type))
+					A.on_varedit(variable)
+
+		else if(istype(O,/turf))
+			for(var/turf/A in world)
+				if(istype(A,O.type))
+					A.on_varedit(variable)
```
Comment (MrPerson) [inline code/modules/admin/verbs/massmodvar.dm]: this might be faster as for(vat/turf/A in block(locate(1,1,1),locate(world.maxx,world.maxy,world.maxz)))

Comment (duncathan) [inline code/modules/admin/verbs/massmodvar.dm]: Doesn't block only return a list of turfs?
If so, you could optimize that further like this:
`for(var/A in block(locate(1,1,1),locate(world.maxx,world.maxy,world.maxz)))`
That way you don't have the redundant type check when getting the list.

**Comment by Jordie0608** [comment]
Won't that cause strange behavior where a user isn't denied changing a var until after they've input everything?

File: code/modules/admin/verbs/massmodvar.dm (hunk 1 above)
Comment (YotaXP) [inline code/modules/admin/verbs/massmodvar.dm]: I would be quite surprised if either of these methods were faster.

Comment (duncathan) [inline code/modules/admin/verbs/massmodvar.dm]: Are you kidding @YotaXP? Looping through world is slow as shit.

Comment (YotaXP) [inline code/modules/admin/verbs/massmodvar.dm]: Doesn't BYOND handle this type of loop in a special way, though?  Or maybe I'm thinking of `for(var/turf/T)`, without the `in world`.  Guess it's worth testing.

Comment (RemieRichards) [inline code/modules/admin/verbs/massmodvar.dm]: ```for(var/turf/T)``` IS ```for(var/turf/T in world)```

Comment (YotaXP) [inline code/modules/admin/verbs/massmodvar.dm]: Dang.  @duncathan is right!

https://paste.ee/p/j8M4I

Comment (RemieRichards) [inline code/modules/admin/verbs/massmodvar.dm]: @YotaXP 

>5 calls
>indicative of anything

1 million Yota, 1 million bare minimum.

Comment (YotaXP) [inline code/modules/admin/verbs/massmodvar.dm]: But each call takes like 5 seconds. :cry:

Comment (RemieRichards) [inline code/modules/admin/verbs/massmodvar.dm]: halve the number then, and leave it running.

**Comment by Aranclanos** [comment]
oh ok then don't
unless you still want to do it before all the input in a different proc because its cool but unrelated to this pr

**Comment by Jordie0608** [comment]
I'd like to touch messy varedit code as little as possible.

File: code/modules/admin/verbs/massmodvar.dm (hunk 1 above)
Comment (Jordie0608) [inline code/modules/admin/verbs/massmodvar.dm]: So which of these suggestions am I using.

**Comment by MrStonedOne** [comment]
Next thing is have a way to have vv tell an object vv was opened on it,

then that object can tell vv what to put in the drop down, and what vars to show and what vars to allow editing of

**Comment by YotaXP** [comment]
@MrStonedOne If the object isn't asked on-change, then you could spoof to bypass that.

**Comment by MrStonedOne** [comment]
well, yeah, obviously... the idea is to allow vv to gray out the edit links

**Comment by duncathan** [comment]
only thing bad in here is mass modify but obviously that's not exactly avoidable

**Pull request merged**